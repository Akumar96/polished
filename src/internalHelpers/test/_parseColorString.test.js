import parseColorString from '../_parseColorString'

describe('parseColorString', () => {
  it('should parse a hex color representation', () => {
    expect(parseColorString('#Ff43AE')).toMatchSnapshot()
  })

  it('should parse a reduced hex color representation', () => {
    expect(parseColorString('#45a')).toMatchSnapshot()
  })

  it('should parse a rgba color representation', () => {
    expect(parseColorString('rgba(174,67,255,0.6)')).toMatchSnapshot()
  })

  it('should parse a rgb color representation', () => {
    expect(parseColorString('rgb(174,67,255)')).toMatchSnapshot()
  })

  it('should throw an error if an invalid color string is provided', () => {
    expect(() => { parseColorString('(174,67,255)') }).toThrow('Couldn\'t parse the color string. Please provide the color in hex, rgb, rgba, hsl or hsla notation as a string.')
  })
})
