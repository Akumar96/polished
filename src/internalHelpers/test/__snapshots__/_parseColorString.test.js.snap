exports[`parseColorString should parse a hex color representation 1`] = `
Object {
  "blue": 174,
  "green": 67,
  "red": 255,
}
`;

exports[`parseColorString should parse a reduced hex color representation 1`] = `
Object {
  "blue": 170,
  "green": 85,
  "red": 68,
}
`;

exports[`parseColorString should parse a rgb color representation 1`] = `
Object {
  "blue": 255,
  "green": 67,
  "red": 174,
}
`;

exports[`parseColorString should parse a rgba color representation 1`] = `
Object {
  "alpha": 0.6,
  "blue": 255,
  "green": 67,
  "red": 174,
}
`;
